src/vs/workbench/api/node/extHost.api.impl.ts

```diff
@@ -258,7 +258,7 @@ export function createApiFactory(
 				return extHostClipboard;
 			},
 			get shell() {
-				return extHostTerminalService.getDefaultShell(configProvider);
+				return extHostTerminalService.getDefaultShell(configProvider, undefined);
```

Comment:
Actually we need this to be synchronous, so let's create the resolver and hang onto it in `ExtHostTerminalService.ctor`